### ISSUE TITLE
upx-3.93 -d cannot decompress its own files

### ISSUE THREAD

**Issue by cieplutki** [upx/upx#91]
After compressing p7zip-16.02 executables I tried to do something (I had them installed upx-compressed and had no intention to compile them again) and tried to decompress and got following error:
```
$ upx -d 7z.so
(...)
upx: 7z.so: CantUnpackException: corrupt b_info
```
program seems to work but 7z.so is a plugin for 7z supporting some exotic packers not "embedded" in 7z/7za and I didn't have real occasion to thoroughly test it. With 7z/zip/bz2 files it seems to work but god one knows what this corruption means and what to expect.

This is what upx'ed programs look like in objdump:
```
$ objdump -x -f -h -p ./7z.so

./7z.so:     file format elf32-i386
./7z.so
architecture: i386, flags 0x00000140:
DYNAMIC, D_PAGED
start address 0x0007cc20

Program Header:
    LOAD off    0x00000000 vaddr 0x00000000 paddr 0x00000000 align 2**12
         filesz 0x00121e26 memsz 0x00121e26 flags r-x
    LOAD off    0x00122888 vaddr 0x00249888 paddr 0x00249888 align 2**12
         filesz 0x0000cc58 memsz 0x00015738 flags rw-
 DYNAMIC off    0x0012ce94 vaddr 0x00253e94 paddr 0x00253e94 align 2**2
         filesz 0x000000e8 memsz 0x000000e8 flags rw-
EH_FRAME off    0x000d1760 vaddr 0x001f7760 paddr 0x001f7760 align 2**2
         filesz 0x0000a2a4 memsz 0x0000a2a4 flags r--
   STACK off    0x00000000 vaddr 0x00000000 paddr 0x00000000 align 2**4
         filesz 0x00000000 memsz 0x00000000 flags rw-

Sections:
Idx Name          Size      VMA       LMA       File off  Algn
SYMBOL TABLE:
no symbols

```

I guess it should be able to decompress properly it's own files.

just in case I attached my executables.

// my environment details:
upx-3.93 (binary from github)
GNU/Linux kernel-3.8.2  i686 Genuine Intel CPU 

PS
I also tried to compile upx form sources but got stuck with this error:

```
g++ -O2 -$(LOT-OF-FLAGS) $(OBJS) -lucl -lz
./../src/stub/scripts/check_whitespace.sh ./..
ERROR: binary file detected ./src/c_file.o: ELF�4(      GCC: (GNU) 4.7.2.symtab.strtab.shstrtab.text.data.bss.comment.note.GNU-stack!44,045FFE�pd
                                                                                                                                                 ��c_file.cppmake: *** [upx.out] Error 1
make: *** Deleting file `upx.out'

```
[p7zip-16.02-lzip-orig.zip](https://github.com/upx/upx/files/947719/p7zip-16.02-lzip-orig.zip)
[p7zip-16.02-lzip-upx-3.93.zip](https://github.com/upx/upx/files/947718/p7zip-16.02-lzip-upx-3.93.zip)

**Comment by markus-oberhumer** [comment]
BTW, for compiling from source please see #64 .

**Comment by cieplutki** [comment]
After that tip with `CHECK_WHITESPACE` I eventually successfully compiled the program but the problem with `upx393: 7z.so: CantUnpackException: corrupt b_info` still persists. Any advice what to do with it? As I don't want to corrupt some program using shared objects/plugins without being able to get back to original.

**Comment by jreiser** [comment]
Fixed by commit d97ff8bb on 2017-03-31 and commit 23e3814f on 2017-03-27.  See also commit 024d0811 on 2017-03-27 (issue 73) and commit 543fa3ae on 2017-03-25.
